libvast/vast/type.hpp

```diff
@@ -638,15 +639,24 @@ struct record_type final : recursive_type<record_type> {
   /// @returns The offsets of fields matching *key*.
   std::vector<offset> find_suffix(std::string_view key) const;
 
-  /// Retrieves the type at a given key.
+  /// Retrieves the field at a given key.
   /// @param key The key to resolve.
-  /// @returns The type at key *key* or `nullptr` if *key* doesn't resolve.
-  const type* at(std::string_view key) const;
+  /// @returns The field at key *key* or `nullptr` if *key* doesn't resolve.
+  const record_field* at(std::string_view key) const;
```

Comment:
For this and the other `at` function, please add these overloads:

```cpp
const record_field* at(std::string_view) && = delete;
const record_field* at(const offset&) && = delete;
```

Reply:
I opted to restrict the existing functions to `... const&` instead.

Reply:
I don't think that'll work, as an xvalue can bind to a const lvalue reference. See here for a small example: https://godbolt.org/z/eb1EoY

Reply:
Thanks, I forgot that this would happen. The explicit overloads are in now.